src/main/java/org/purpurmc/purpurextras/modules/BannerLayersModule.java

```diff
@@ -0,0 +1,168 @@
+package org.purpurmc.purpurextras.modules;
+
+import org.bukkit.Material;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.event.inventory.InventoryAction;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.LoomInventory;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.inventory.meta.BannerMeta;
+import org.purpurmc.purpurextras.PurpurConfig;
+import org.purpurmc.purpurextras.PurpurExtras;
+
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * If enabled, banners will be able to use more than the limit of 6 layers.
+ * In default config up to 10 banner layers are allowed.
+ */
+public class BannerLayersModule implements PurpurExtrasModule, Listener {
+    protected BannerLayersModule() {
+        PurpurConfig config = PurpurExtras.getPurpurConfig();
+        Map<String, Object> defaults = new HashMap<>();
+
+        defaults.put("layers", 10);
+        ConfigurationSection section = config.getConfigSection("settings.banner-layers", defaults);
+        Logger logger = PurpurExtras.getInstance().getLogger();
+        if (section.getInt("layers") < 6) {
+            logger.warning("Banner layers is set to less than 6, this is not currently supported.");
+        }
+    }
+
+    // predefined arraylist
+    private static final List<Material> bannertypes = new ArrayList<>();
```

Comment:
This should either be a Set for faster .contains, or just use vanilla or paper type tag instead, the latter would be preferred

Reply:
How do i do the vanilla/paper type tag?

Reply:
https://purpurmc.org/javadoc/com/destroystokyo/paper/MaterialSetTag.html